src/libsyntax/parse/attr.rs

```diff
@@ -18,6 +23,34 @@ const DEFAULT_UNEXPECTED_INNER_ATTR_ERR_MSG: &str = "an inner attribute is not \
                                                      permitted in this context";
 
 impl<'a> Parser<'a> {
+    crate fn parse_arg_attributes(&mut self) -> PResult<'a, Vec<ast::Attribute>> {
+        let mut attrs: Vec<ast::Attribute> = Vec::new();
+        let mut diagnostic = self.fatal("allow, cfg, cfg_attr, deny, forbid and warn \
+            are the only allowed built-in attributes in function arguments");
+        loop {
+            debug!("parse_arg_attributes: self.token={:?}", self.token);
+            match self.token {
+                token::Pound => {
+                    let attr = self.parse_attribute(false)?;
+                    if self.has_valid_arg_attribute(&attr) {
+                        attrs.push(attr);
+                    }
+                    else {
+                        diagnostic.span_note(attr.span, "forbidden built-in attribute");
```

Comment:
This is not a parsing error, it's a semantic error that should be checked e.g. in `ast_validation.rs`.